torch/linalg/__init__.py

```diff
@@ -140,3 +140,43 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorinv = _add_docstr(_linalg.linalg_tensorinv, r"""
+linalg.tensorinv(input, ind=2, *, out=None) -> Tensor
+
+Computes a tensor ``input_inv`` such that ``tensordot(input_inv, input, ind) == I_n``,
+where ``I_n`` is the n-dimensional identity tensor and ``n`` is equal to ``input.ndim``.
+The resulting tensor ``input_inv`` has the shape equal to ``input.shape[ind:] + input.shape[:ind]``.
+
+Supports real and, only on the CPU, complex inputs.
+
+.. note:: If :attr:`input` is not a 'square' tensor, meaning it does not satisfy the requirement
+          ``prod(input.shape[ind:]) == prod(input.shape[:ind])`` or if :attr:`input` is not invertible,
+          then a RuntimeError will be thrown.
+
+.. note:: If :attr:`input` is a 2-dimensional tensor and ``ind=1``, then this function computes matrix inverse.
+
+Args:
+    input (Tensor): The tensor to invert. Its shape must satisfy ``prod(input.shape[:ind]) == prod(input.shape[ind:])``.
+    ind (int): The positive integer representing how many first indices are involved in the inverse contraction.
```

Comment:
"The positive" -> "A positive"

Reply:
I agree that it's better to avoid "squareness", not to confuse people. What is the inverse equation, then?
I think I'll change it to "A positive integer that affects the result's shape.  Default: 2." And how it affects the shape is described in the main paragraph.

Reply:
That's OK. You're right that my own suggestion was a little vague. By "inverse equation" I meant the tensordot expression in the op's description. Maybe we can reference that? I like your suggestion but also find "affects the result's shape" to be a little vague.